apps/stripe/src/__tests__/integration/webhooks/stripe-webhook.test.ts

```diff
@@ -0,0 +1,230 @@
+      },
+    }),
+  );
+  afterAll(() => mswServer.close());
+  afterEach(() => mswServer.resetHandlers());
+
+  beforeEach(async () => {
+    await apl.set({
+      saleorApiUrl: realSaleorApiUrl,
+      appId: mockedSaleorAppId,
+      token: "mocked-token",
+      jwks: "{}",
+    });
+
+    await configRepo.saveStripeConfig({
+      saleorApiUrl: realSaleorApiUrl,
+      appId: mockedSaleorAppId,
+      config: StripeConfig.create({
+        publishableKey: createStripePublishableKey(env.INTEGRATION_STRIPE_PK)._unsafeUnwrap(),
+        name: "Config name",
+        webhookId: "we_123",
+        restrictedKey,
+        webhookSecret: mockStripeWebhookSecret,
+        id: configId,
+      })._unsafeUnwrap(),
+    });
+
+    await configRepo.updateMapping(
+      {
+        saleorApiUrl: realSaleorApiUrl,
+        appId: mockedSaleorAppId,
+      },
+      {
+        configId: configId,
+        channelId: mockedSaleorChannelId,
+      },
+    );
+
+    const paymentIntentResult = await paymentIntentApi.createPaymentIntent({
+      stripeMoney,
+      idempotencyKey: new RandomId().generate(),
+      intentParams: {
+        return_url: "https://saleor-stripe-integration-test.com",
+        automatic_payment_methods: {
+          enabled: true,
+        },
+        payment_method: "pm_card_visa",
+        confirm: true,
+      },
+    });
+
+    stripePaymentIntentId = createStripePaymentIntentId(paymentIntentResult._unsafeUnwrap().id);
+
+    await transactionRecorderRepo.recordTransaction(
+      {
+        saleorApiUrl: realSaleorApiUrl,
+        appId: mockedSaleorAppId,
+      },
+      new RecordedTransaction({
+        saleorTransactionId: mockedSaleorTransactionIdBranded,
+        stripePaymentIntentId,
+        saleorTransactionFlow: createSaleorTransactionFlow("CHARGE"),
+        resolvedTransactionFlow: createResolvedTransactionFlow("CHARGE"),
+        selectedPaymentMethod: "card",
+      }),
+    );
+  });
+
+  it("should successfully process a payment intent event", async () => {
+    const payload = paymentIntentSucceededEventFixture(stripePaymentIntentId);
+
+    const payloadString = JSON.stringify(payload, null, 2);
+
+    const stripeClient = StripeClient.createFromRestrictedKey(restrictedKey);
+
+    const header = stripeClient.nativeClient.webhooks.generateTestHeaderString({
+      payload: payloadString,
+      secret: mockStripeWebhookSecret,
+    });
+
+    await testApiHandler({
+      appHandler: stripeWebhookHandlers,
+      url: `/api/webhooks/stripe?configurationId=${configId}&saleorApiUrl=${encodeURIComponent(
+        realSaleorApiUrl.toString(),
+      )}&appId=${mockedSaleorAppId}`,
+      async test({ fetch }) {
+        const response = await fetch({
+          method: "POST",
+          body: payloadString,
+          headers: new Headers({
+            "stripe-signature": header,
+          }),
+        });
+
+        const body = await response.text();
+
+        expect(body).toStrictEqual("Ok");
```

Comment:
I don't think this is enough of an assertion. We should test visible effects of the webhooks. Response to Stripe is one of them, but in the meantime we should at least check if Saleor transactionEventReport was called (so we check how it *integrates* with external systems)